File: 10 - Is/is.c
#include <stdio.h>
#define TAM 200
/*
 * X1 - recebe uma String(ve se ela só tem vogais) se sim,retorna verdadeiro
 * X2 - recebe uma String(ve se ela só tem consoantes) se sim,retorna verdadeiro
 * X3 - recebe uma String(ve se ela é um numero inteiro) se sim,retorna verdadeiro
 * X4 - recebe uma String(ve se ela á um numero real) se sim,retorna verdadeiro
*/

//X1(true) - so pode estar contido em (a e i o u)[e maiusculos]
//X2(true) - deve estar entre (A e Z),mas diferentes de (a e i o u)[e maiusculos]
//X3(true) - so pode ser numero
//X4(true) - so numero e no maximo um '.' ou uma ',';

int tamanho(char str[]){
//-- variaveis --//
int i = 0;
	while(str[i] != '\0'){
	i++;
	}
	return i;
}

void lowkey(char str[]){
//-- variaveis --//
int i = 0;

	while(str[i] != '\0'){
		if(str[i] >= 'A' && str[i] <= 'Z'){
		str[i] = str[i] + ('a' - 'A');
		}
    i++;
	}
}

void X1X2(char str[],int tamanho){
//-- variaveis --//
int vogal = 0;
int consoante = 0;
int i = 0;

lowkey(str);

	while(str[i] != '\0'){
	if((str[i] >= 'a' && str[i] <= 'z') && (str[i] == 'a'||str[i] == 'e'||str[i] == 'i'||str[i] == 'o'||str[i] == 'u')){
	vogal++;
		}
	else{
	if(str[i] >= 'a' && str[i] <= 'z'){
    consoante++;
    }

	}
		i++;
	}


//-- if vogal e consoante --//
if(vogal > 0 && vogal == tamanho){
	printf("SIM ");
	}
else{
	printf("NAO ");
	}
if(consoante >  0 && consoante == tamanho){
	printf("SIM ");
	}
else{
	printf("NAO ");
}
}


void X3(char str[], int tamanho){
//-- variaveis --//
int i = 0;

	while((str[i] != '\0') && (str[i] >= '0' && str[i] <= '9')){
	i++;
	}
	if(i == tamanho){
		printf("SIM ");
	}
	else{
		printf("NAO ");
	}

}
void X4(char str[],int tamanho){
//-- variaveis --//
int i = 0;
int point = 0;

	while((str[i] != '\0') && (str[i] >= '0' && str[i] <= '9'||str[i] == '.'||str[i] == ',')){
	if(str[i] == '.' || str[i] == ','){
	point++;
	}
    i++;
	}
	if(i == tamanho && point <= 1){
		printf("SIM ");
	}
	else{
		printf("NAO ");
	}	
}

int main(){
//-- variaveis --//
char str[TAM];
int tam = 0;

//-- pede a string --//
fgets(str,TAM,stdin);

	while(!(str[0] == 'F' && str[1] == 'I' && str[2] == 'M')){
tam = tamanho(str);

if(str[tam - 1] == '\n'){
    str[tam - 1] = '\0';
    tam--;
}

//-- metodos --//
X1X2(str,tam);
X3(str,tam);
X4(str,tam);
printf("\n");
	fgets(str,TAM,stdin);
	}
}

File: java/Is.java
import java.util.Scanner;

public class Is {
    public static void main (String[] args){
    Scanner scan = new Scanner(System.in);
    
    //-- pede uma string --//
    String str = scan.nextLine();
    
    while(!(str.equals("FIM"))){

    //-- metodos --//
    X1X2(str);
    X3(str);
    X4(str);
    
    System.out.println("");
    str = scan.nextLine();

    }
        scan.close();
    }

    public static String lowkey(String str, int i){
    char[] chars = str.toCharArray();
    lowkeyRec(chars, i);
    return new String(chars);
}

    private static void lowkeyRec(char[] chars, int i){
    if(i == chars.length){
        return; // condição de parada
    }
    if(chars[i] >= 'A' && chars[i] <= 'Z'){
        chars[i] = (char)(chars[i] + ('a' - 'A'));
    }
    lowkeyRec(chars, i+1);
    }

    public static void ContaVoCO(String str, int i, int[] vogal, int[] consoante){
        if(i == str.length()){
            return;
        }
        if((str.charAt(i) >= 'a' && str.charAt(i) <= 'z') && (str.charAt(i)=='a'||str.charAt(i)=='e'||str.charAt(i)=='i'||str.charAt(i)=='o'||str.charAt(i)=='u')){
            vogal[0]++;
        }
        else if(str.charAt(i) >= 'a' && str.charAt(i) <= 'z'){
            consoante[0]++;
        }
        ContaVoCO(str,i+1,vogal,consoante);
    }

    public static void X1X2(String str){
        int[] vogal = new int[1];
        int[] consoante = new int[1];

        lowkey(str,0);
        ContaVoCO(str,0,vogal,consoante);

        //-- verifica vogal e consoante --//
        if(vogal[0] > 0 && vogal[0] == str.length()){
            System.out.print("SIM ");
            }
        else{
            System.out.print("NAO ");
            }
        if(consoante[0] >  0 && consoante[0] == str.length()){
            System.out.print("SIM ");
            }
        else{
            System.out.print("NAO ");
            }
    }

    //-- verifica se é numero real --//
    public static int verificadorReal(String str, int i, int point){
        if(i == str.length()){
            return i;
        }
        char c = str.charAt(i);
        if(!(c >= '0' && c <= '9') && !(c == '.' || c == ',')){
            return i;
        }
        if(c == '.' || c == ','){
            point++;
        }
        if(point > 1){
            return i;
        }
    return verificadorReal(str, i + 1, point);
    }

    public static int verificadorInt(String str, int i){
        if(i == str.length()){
            return i;
        }
        char c = str.charAt(i);
        if(!(c >= '0' && c <= '9')){
            return i;
        }
        return verificadorInt(str, i + 1);
    }
    public static void X3(String str){
        int i = verificadorInt(str,0);
        if(i == str.length()){
            System.out.print("SIM ");
        }
        else{
            System.out.print("NAO ");
        }
    }

    public static void X4(String str){
        int point = 0;
        int i = verificadorReal(str, 0, point);
        if(i == str.length() && point <= 1){
            System.out.print("SIM");
        }
        else{
            System.out.print("NAO");
        }   
    }


}
